w3lib/url.py

```diff
@@ -88,9 +95,29 @@ def safe_url_string(  # pylint: disable=too-many-locals
     path_encoding: str = "utf8",
     quote_path: bool = True,
 ) -> str:
-    """Convert the given URL into a legal URL by escaping unsafe characters
-    according to RFC-3986. Also, ASCII tabs and newlines are removed
-    as per https://url.spec.whatwg.org/#url-parsing.
+    """Return a URL equivalent to *url* that a wide range of web browsers and
+    web servers consider valid.
+
+    *url* is parsed according to the rules of the `URL living standard`_,
+    and during serialization additional characters are percent-encoded to make
+    the URL valid by additional URL standards.
+
+    .. _URL living standard: https://url.spec.whatwg.org/
+
+    The returned URL is valid by *all* of the following URL standards known to
```

Comment:
@Gallaecio I think you mentioned that the implementation is not 100% compatible, until we solve https://github.com/scrapy/w3lib/issues/204. Is there some detail to it?

Reply:
No, there is not. This is why I said:

> It describes more what the functions should (in my opinion) do than what they actually do.

Any suggestion on how to handle this? Should we say that the function *aims* to do this, but there are know issues or divergences at the moment?

Reply:
A small language tweak ("should be valid" instead of "is valid") might be enough.